### PR TITLE
[MainUI] Add tags & script language to script settings

### PR DESCRIPTION
**Pull request by florian-h05**
Reference #1597.
Reference https://github.com/openhab/openhab-addons/issues/14005.

## Description

This PR extends the script settings to allow the user to set tags and change the MIME type after creation of the script.
The ability to add and edit tags was requested several times (I think also by @rkoshak).

The ability to change the MIME type of a script after creation is **very important for openHAB 4**, because Nashorn will (very likely) swap it's MIME type with GraalJS, but Nashorn scripts do not run on GraalJS without some changes. With the extended script settings, the user can just change the MIME type to the new MIME type of Nashorn.

The "Script" tag that all scripts have is hidden in the settings.

### Script Creation

image

Note that the "Tags" section is newly added and doesn't show the "Script" tag although it is present.

### Editing a Script

Here has already been a settings popup. It has been extended with tags and the ability to change the MIME type:
image

Note that this script settings are from a Blockly script, therefore only Javascript versions (I've only installed GraalJS) are listed. If you open these settings for a normal script, you can change the MIME type to any of the available engines, e.g. Rules DSL.

### Editing a Script in a Rule

The settings are newly introduced for this case and limited to the MIME type:
image

/ping @ghys @rkoshak @stefan-hoehn

### CODE REVIEW

**Comment by stefan-hoehn** [comment]
I love the idea. Again a step forward to easier allow to the new ECMAScript version.

regarding the tags:

I recently added some more search criteria in /settings/rules/ (see https://github.com/openhab/openhab-webui/pull/1540)
Maybe it is possible to show and search for the tags there (like in /settings/items/) as well?

**Comment by florian-h05** [comment]
I've just checked it:
- searching by tags works
- searching by script states works
- searching by UID works
- searching by description works

FYI: The code to list rules and the one to list scripts is the same and scripts are in fact only a special type of rules.

**Comment by florian-h05** [comment]
@ghys This is now ready for review, it's working fine on my system and doesn't print any errors to the console.

**Comment by stefan-hoehn** [comment]
Does it not only search for the tags but also show the tags?. Our idea until now was that everything that is searchable is also shown on the list and vice versa.

**Comment by florian-h05** [comment]
It also shows the tags -- it behaves exactly the same like the rule overview.

File: bundles/org.openhab.ui/web/src/pages/settings/rules/script/script-edit.vue [hunk 1]
```diff
@@ -243,6 +231,22 @@ export default {
         this.load()
       })
     },
+    loadScriptModules () {
+      this.$oh.api.get('/rest/module-types/script.ScriptAction').then((data) => {
```
Comment (ghys) [inline bundles/org.openhab.ui/web/src/pages/settings/rules/script/script-edit.vue:235]: One tiny general remark:
this component is also used to edit script conditions.
I understand you probably took this into account and figured there was no reason the list of languages would be different from the action to the condition, so good enough, but just pointing it out.

Comment (florian-h05) [reply bundles/org.openhab.ui/web/src/pages/settings/rules/script/script-edit.vue:235]: Thanks for that information. 
I wasn‘t aware of this, but as there is no difference in the list of available languages for actions and conditions, this should be fine.
If I see it correctly, this component also did not take into account whether it edits an action or a condition before this PR.

**Pull request merged**